modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -5186,6 +5186,28 @@ bool GDScriptAnalyzer::is_type_compatible(const GDScriptParser::DataType &p_targ
 	return false;
 }
 
+bool GDScriptAnalyzer::is_super_type(const GDScriptParser::DataType &left_type, const GDScriptParser::DataType &right_type) {
```

Comment:
This is a "best effort" implementation as suggested originally by @dalexeev - while this is far from perfect it definitely gets the job done.